### PR TITLE
feat: admin/moderator permissions

### PR DESCRIPTION
**Pull request by sahas-01**
Added permission for moderators/admins(specific addresses) to do the following:
- delete comments
- delete replies
- delete posts

### CODE REVIEW

File: src/util/deleteContentByMod.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { MessageType } from 'antd/es/message/interface';
+import { NotificationStatus } from '~src/types';
+import queueNotification from '~src/ui-components/QueueNotification';
+import nextApiClientFetch from './nextApiClientFetch';
+import { ProposalType } from '~src/global/proposalType';
+
+export async function deleteContentByMod(postId:number|string|undefined, proposalType:ProposalType, reason:string, commentId?:string, replyId?:string, onDeleteComment?:()=>void , onDeleteReply?:()=>void, onDeletePost?:()=>void)   {
```
Comment (alphainfinitus) [inline src/util/deleteContentByMod.ts:11]: postId shouldn't be undefined

Comment (alphainfinitus) [reply src/util/deleteContentByMod.ts:11]: `onDeleteComment?:()=>void` , `onDeleteReply?:()=>void`, `onDeletePost?:()=>void` can be changed to a single `onSuccess()`

Comment (sahas-01) [reply src/util/deleteContentByMod.ts:11]: Resolved, pls check

File: pages/api/v1/auth/actions/deleteContentByMod.ts
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/deleteContentByMod.ts:1]: please use messages from the message file

File: pages/api/v1/auth/actions/deleteContentByMod.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			isDelete: true
+		});
+		return res.status(200).json({ message: 'Reply deleted.' });
+	}
+	if(postId && commentId && !replyId){
+		ref = ref
+			.collection('comments')
+			.doc(String(commentId));
+		await ref.update({
+			isDelete: true
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/deleteContentByMod.ts:48]: key should be `isDeleted`

File: pages/api/v1/auth/actions/deleteContentByMod.ts
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/deleteContentByMod.ts:1]: ref assigning flow can be optimised so that ref.update is only written once

File: pages/api/v1/listing/off-chain-posts.ts [hunk 3]
```diff
@@ -98,6 +98,7 @@ export async function getOffChainPosts(params: IGetOffChainPostsParams) : Promis
 						comments_count: commentsQuerySnapshot.data()?.count || 0,
 						created_at: created_at?.toDate? created_at?.toDate(): created_at,
 						gov_type:docData?.gov_type ,
+						isDelete: docData?.isDelete || false,
```
Comment (alphainfinitus) [inline pages/api/v1/listing/off-chain-posts.ts:101]: `isDeleted`

File: pages/api/v1/posts/on-chain-post.ts [hunk 4]
```diff
@@ -365,6 +365,7 @@ export async function getComments(commentsSnapshot: FirebaseFirestore.QuerySnaps
 				created_at: data.created_at?.toDate ? data.created_at.toDate(): data.created_at,
 				history: history,
 				id: data.id,
+				isDelete: data.isDelete || false,
```
Comment (alphainfinitus) [inline pages/api/v1/posts/on-chain-post.ts:368]: `isDeleted`

File: src/util/deleteContentByMod.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	if (deleteError || !deleteData) {
+		console.error('Error deleting content: ', deleteError);
+		queueNotification({
+			header: 'Error!',
+			message: deleteError || 'Error in deleting content',
+			status: NotificationStatus.ERROR
+		});
+	}
+
+	if(deleteData && commentId && replyId){
```
Comment (alphainfinitus) [inline src/util/deleteContentByMod.ts:28]: onSuccess?.(); should be called if `deleteData` is present

File: src/util/deleteContentByMod.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+export async function deleteContentByMod(postId:number|string, proposalType:ProposalType, reason:string, commentId?:string, replyId?:string, onSuccess?:()=>void)   {
+	const { data: deleteData , error: deleteError } = await nextApiClientFetch<MessageType>('api/v1/auth/actions/deleteContentByMod', {
+		commentId,
+		postId : Number(postId),
+		postType: proposalType,
+		reason,
+		replyId
+	});
+	if (deleteError || !deleteData) {
```
Comment (alphainfinitus) [inline src/util/deleteContentByMod.ts:19]: there should be an early return in this block

File: src/components/Listing/OffChain/OffChainPostsListing.tsx [hunk 7]
```diff
@@ -25,7 +25,8 @@ const OffChainPostsListing: FC<IOffChainPostsListingProps> = ({ className, posts
 			</div>
 		);
 	}
-
+	//filter the posts that are deleted
+	posts = posts.filter((post) => post.isDelete !== true || post.isDelete === undefined);
```
Comment (alphainfinitus) [inline src/components/Listing/OffChain/OffChainPostsListing.tsx:29]: please double check this might cause pagination to break, if we use `startAt` or `startAfter`

File: src/components/Post/ActionsBar/ReportButton/index.tsx [hunk 8]
```diff
@@ -153,33 +161,71 @@ const ReportButton: FC<IReportButtonProps> = (props) => {
...
+		setLoading(true);
+		await form.validateFields();
+		const validationErrors = form.getFieldError('reason');
+		if(validationErrors.length > 0) return;
+		setFormDisabled(true);
+		const reason = form.getFieldValue('comments');
+		if(allowed_roles?.includes('moderator')) {
+			if(!postId) return;
+			if(postId && commentId && replyId) {
+				await deleteContentByMod(postId, proposalType, reason, commentId, replyId, onDeleteReply);
```
Comment (alphainfinitus) [inline src/components/Post/ActionsBar/ReportButton/index.tsx:174]: same line of code three times, the last param can be changed

File: pages/api/v1/auth/actions/deleteContentByMod.ts (hunk 2 above)
Comment (sahas-01) [reply pages/api/v1/auth/actions/deleteContentByMod.ts:48]: resolved, pls check

File: pages/api/v1/auth/actions/deleteContentByMod.ts
Comment (sahas-01) [reply pages/api/v1/auth/actions/deleteContentByMod.ts:1]: updated the flows, pls check

File: pages/api/v1/listing/off-chain-posts.ts (hunk 3 above)
Comment (sahas-01) [reply pages/api/v1/listing/off-chain-posts.ts:101]: resolved,pls check

File: pages/api/v1/posts/on-chain-post.ts (hunk 4 above)
Comment (sahas-01) [reply pages/api/v1/posts/on-chain-post.ts:368]: resolved,pls check

File: src/auth/auth.ts [hunk 9]
```diff
@@ -111,6 +111,7 @@ class AuthService {
 			web3_signup: web3signup
 		};
 		const newUserRef = firebaseAdmin.firestore().collection('users').doc(userId);
+		console.log('newUserRef', newUserRef);
```
[the remark this answers is not in the document]
Comment (sahas-01) [reply src/auth/auth.ts:114]: removed, pls check

File: src/components/Post/Comment/Comment.tsx [hunk 10]
```diff
@@ -35,6 +35,7 @@ export interface IComment {
   profile?:any;
 	post_index?: number;
 	post_type?: string;
+	isDelete?: boolean;
```
Comment (alphainfinitus) [inline src/components/Post/Comment/Comment.tsx:38]: `isDeleted`

File: src/components/Post/Comment/CommentsContainer.tsx [hunk 11]
```diff
@@ -77,7 +77,7 @@ interface ISentimentsPercentage {
 const getSortedComments = (comments: {[index:string]:Array<IComment>}) => {
 	const commentResponse:any = {};
 	for(const key in comments){
-		commentResponse[key] = comments[key].sort((a, b) => (dayjs(a.created_at).diff(dayjs(b.created_at))));
+		commentResponse[key] = comments[key].filter(comment => comment.isDelete?false:true).sort((a, b) => (dayjs(a.created_at).diff(dayjs(b.created_at))));
```
Comment (alphainfinitus) [inline src/components/Post/Comment/CommentsContainer.tsx:80]: `isDeleted`

File: src/components/Post/Comment/CommentsContainer.tsx [hunk 12]
```diff
@@ -102,7 +102,7 @@ const CommentsContainer: FC<ICommentsContainerProps> = (props) => {
 	const [filterSentiments, setFilterSentiments] = useState<ESentiments|null>(null);
 	const router = useRouter();
 	let allComments = Object.values(comments)?.flat() || [];
-	const allCommentsLength = timelines.reduce((a, b) => a + b.commentsCount, 0);
+	// const allCommentsLength = timelines.reduce((a, b) => a + b.commentsCount, 0);
```
Comment (alphainfinitus) [inline src/components/Post/Comment/CommentsContainer.tsx:105]: unnecessary line

File: pages/api/v1/auth/actions/deleteContentByMod.ts [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+	postType: ProposalType;
+}
+
+async function handler(req: NextApiRequest, res: NextApiResponse<MessageType>) {
+	if (req.method !== 'POST') return res.status(405).json({ message: 'Invalid request method, POST required.' });
+
+	const network = String(req.headers['x-network']);
+	if(!network) return res.status(400).json({ message: 'Missing network name in request headers' });
+
+	if(network !== 'kusama' && network !== 'polkadot') return res.status(400).json({ message: 'Invalid network in request header' });
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/deleteContentByMod.ts:32]: it should be an or condition instead of `&&`

File: pages/api/v1/auth/actions/deleteContentByMod.ts [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+	}
+	if(postId && commentId && !replyId){
+		ref = ref
+			.collection('comments')
+			.doc(String(commentId));
+	}
+	await ref.update({
+		isDeleted: true
+	});
+	const args:Args = {
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/deleteContentByMod.ts:61]: notificationArgs

File: pages/api/v1/auth/actions/editPost.ts
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/editPost.ts:1]: isDeleted needs to be added in the batch call as well

File: pages/api/v1/listing/off-chain-posts.ts [hunk 15]
```diff
@@ -193,6 +195,7 @@ export async function getOffChainPosts(params: IGetOffChainPostsParams) : Promis
 			status: 200
 		};
 	} catch (error) {
+		console.log('error', error);
```
Comment (alphainfinitus) [inline pages/api/v1/listing/off-chain-posts.ts:198]: remove console log please

File: pages/api/v1/listing/off-chain-posts.ts [hunk 16]
```diff
@@ -63,19 +63,21 @@ export async function getOffChainPosts(params: IGetOffChainPostsParams) : Promis
...
+				.where('isDeleted','==',false)
 				.orderBy(orderedField, order)
 				.limit(Number(listingLimit) || LISTING_LIMIT)
 				.offset((Number(page) - 1) * Number(listingLimit || LISTING_LIMIT))
 				.get();
 
 		const count = (filterBy && filterBy.length > 0)
-			?(await offChainCollRef.where('tags','array-contains-any',filterBy).count().get()).data().count
-			: (await offChainCollRef.count().get()).data().count;
+			?(await offChainCollRef.where('tags','array-contains-any',filterBy).where('isDeleted','==',false).count().get()).data().count
```
Comment (alphainfinitus) [inline pages/api/v1/listing/off-chain-posts.ts:79]: please check if this needs an index

File: pages/api/v1/posts/comments/getCommentsByTimeline.ts
Comment (alphainfinitus) [inline pages/api/v1/posts/comments/getCommentsByTimeline.ts:1]: isDeleted filter needs to be on sentiment query as well

File: pages/api/v1/posts/on-chain-post.ts [hunk 17]
```diff
@@ -379,40 +379,45 @@ export async function getComments(commentsSnapshot: FirebaseFirestore.QuerySnaps
...
-							user_id: user_id,
-							username
-						});
 					}
-				}
-			});
+				});
+			}
+			catch(err){
+				console.log('error in getting replies',err);
```
Comment (alphainfinitus) [inline pages/api/v1/posts/on-chain-post.ts:419]: this try catch block is not needed

File: src/auth/auth.ts [hunk 18]
```diff
@@ -858,18 +858,15 @@ class AuthService {
 
 		let default_address = null;
 		let addresses: string[] = [];
-
```
Comment (alphainfinitus) [inline src/auth/auth.ts:861]: please keep lines between blocks for readability

File: src/components/Listing/OffChain/OffChainPostsListing.tsx [hunk 19]
```diff
@@ -25,7 +25,6 @@ const OffChainPostsListing: FC<IOffChainPostsListingProps> = ({ className, posts
 			</div>
 		);
 	}
-
```
Comment (alphainfinitus) [inline src/components/Listing/OffChain/OffChainPostsListing.tsx:28]: unnecessary change

File: src/components/Post/ActionsBar/ReportButton/index.tsx [hunk 20]
```diff
@@ -33,7 +41,7 @@ const reasons = [
 ];
 
 const ReportButton: FC<IReportButtonProps> = (props) => {
-	const { type, postId, commentId, replyId, className, proposalType } = props;
+	const { type, postId, commentId, replyId, className, proposalType, isDeleteModal , isReply, isPost, onDeleteReply, onDeleteComment , onDeletePost, allowed_roles } = props;
```
Comment (alphainfinitus) [inline src/components/Post/ActionsBar/ReportButton/index.tsx:44]: `onDeleteReply, onDeleteComment , onDeletePost` can be replaced with `onSuccess`

File: src/components/Post/ActionsBar/ReportButton/index.tsx [hunk 21]
```diff
@@ -153,33 +161,70 @@ const ReportButton: FC<IReportButtonProps> = (props) => {
 
 		setLoading(false);
 	};
-
+	const handleDelete = async () => {
+		setLoading(true);
```
Comment (alphainfinitus) [inline src/components/Post/ActionsBar/ReportButton/index.tsx:165]: early return : `if(!allowed_roles?.includes('moderator') || isNaN(Number(postId)) return;`

File: src/components/Post/ActionsBar/ReportButton/index.tsx [hunk 22]
```diff
@@ -153,33 +161,70 @@ const ReportButton: FC<IReportButtonProps> = (props) => {
...
+		if(allowed_roles?.includes('moderator')) {
+			if(isNaN(Number(postId))) return;
+			let onSuccess = onDeletePost;
+			if(commentId) {
+				onSuccess = onDeleteComment;
+			}
+			if(commentId && replyId) {
+				onSuccess = onDeleteReply;
+			}
+			await deleteContentByMod(postId as string | number, proposalType, reason, commentId, replyId, onSuccess);
```
Comment (alphainfinitus) [inline src/components/Post/ActionsBar/ReportButton/index.tsx:180]: setLoading to false after this

File: src/components/Post/Comment/EditableCommentContent.tsx [hunk 23]
```diff
@@ -285,52 +284,58 @@ const EditableCommentContent: FC<IEditableCommentContentProps> = (props) => {
...
+					message: deleteCommentError || 'There was an error in deleting your comment.',
+					status: NotificationStatus.ERROR
+				});
+			}
+			if (data){
+				removeCommentContent();
+			}
+		}
+		else{
+			removeCommentContent();
```
Comment (alphainfinitus) [inline src/components/Post/Comment/EditableCommentContent.tsx:338]: if a user is moderator and clicks on delete (when deleting his own comment), his comment is not deleted herebut only removed from state

File: src/components/Post/Comment/EditableCommentContent.tsx [hunk 24]
```diff
@@ -285,52 +284,58 @@ const EditableCommentContent: FC<IEditableCommentContentProps> = (props) => {
...
+		});
+		queueNotification({
+			header: 'Success!',
+			message: `${allowed_roles?.includes('moderator') && userId! == id? 'The' : 'Your'} comment was deleted.`,
+			status: NotificationStatus.SUCCESS
+		});
+	};
+
+	const deleteComment = async () => {
+		if (!allowed_roles?.includes('moderator')) {
```
Comment (alphainfinitus) [inline src/components/Post/Comment/EditableCommentContent.tsx:319]: this condition is not needed `if (!allowed_roles?.includes('moderator'))`

File: src/components/Post/Comment/EditableCommentContent.tsx (hunk 23 above)
Comment (alphainfinitus) [reply src/components/Post/Comment/EditableCommentContent.tsx:338]: this else clause needs to be removed

File: src/components/Post/Comment/EditableCommentContent.tsx [hunk 25]
```diff
@@ -352,7 +357,13 @@ const EditableCommentContent: FC<IEditableCommentContentProps> = (props) => {
 		id===userId ? {
 			key:4,
 			label:<div className={`flex items-center shadow-none text-[10px] text-slate-400 leading-4 ml-[-1.8px] ${poppins.variable} ${poppins.className} border-none` } onClick={() => {deleteComment();}}><DeleteIcon className='mr-1' />Delete</div>
-		}:null
+		}:
+			allowed_roles?.includes('moderator') && (network.includes('kusama') || network.includes('polkadot')) ?
+				{
+					key: 4,
+					label: <ReportButton isDeleteModal={true} proposalType={postType} className={`flex items-center shadow-none text-slate-400 text-[10px] leading-4 ml-[-7px] h-[17.5px] w-[100%] rounded-none hover:bg-transparent ${poppins.variable} ${poppins.className} `} type='comment' allowed_roles={allowed_roles} onDeleteComment={deleteComment} isReply={false} commentId={commentId} postId={postIndex}/>
```
Comment (alphainfinitus) [inline src/components/Post/Comment/EditableCommentContent.tsx:364]: `onDeleteComment` -> `onSuccess = {removeCommentContent}`

File: src/components/Post/Comment/EditableReplyContent.tsx [hunk 26]
```diff
@@ -240,53 +240,62 @@ const EditableReplyContent = ({ userId, className, commentId, content, replyId ,
...
+				}
+				if(flag){
+					break;
+				}
+			}
+			return comments;
+		});
+		queueNotification({
+			header: 'Success!',
+			message: `${allowed_roles?.includes('moderator')? 'The' : 'Your'} reply was deleted.`,
```
Comment (alphainfinitus) [inline src/components/Post/Comment/EditableReplyContent.tsx:268]: please change message to `The reply has been deleted.`

File: src/components/Post/Comment/EditableReplyContent.tsx
Comment (alphainfinitus) [inline src/components/Post/Comment/EditableReplyContent.tsx:1]: same changes as EditableCommentContent

File: src/components/Post/Tabs/PostDescription.tsx [hunk 27]
```diff
@@ -25,20 +29,76 @@ const CommentsContainer = dynamic(() => import('../Comment/CommentsContainer'),
 });
 
 interface IPostDescriptionProps {
+	allowed_roles?: string[] | null;
```
Comment (alphainfinitus) [inline src/components/Post/Tabs/PostDescription.tsx:32]: allowed_roles should be used from context and not prop drilled

File: src/components/Post/Tabs/PostDescription.tsx [hunk 28]
```diff
@@ -25,20 +29,76 @@ const CommentsContainer = dynamic(() => import('../Comment/CommentsContainer'),
...
 	className?: string;
 	canEdit: boolean | '' | undefined;
 	id: number | null | undefined;
 	isEditing: boolean;
 	isOnchainPost: boolean;
+	trackName?: string;
```
Comment (alphainfinitus) [inline src/components/Post/Tabs/PostDescription.tsx:38]: `trackName` is available in `PostDataContext`

File: src/components/Post/Tabs/PostDescription.tsx [hunk 29]
```diff
@@ -25,20 +29,76 @@ const CommentsContainer = dynamic(() => import('../Comment/CommentsContainer'),
...
+	isOffchainPost?: boolean;
```
[the remark this answers is not in the document]
Comment (alphainfinitus) [reply src/components/Post/Tabs/PostDescription.tsx:39]: `postType` is available in `PostDataContext`

`const isOffChainPost: Boolean  = postType == ProposalType.DISCUSSIONS || postType == ProposalType.GRANTS`

File: src/components/Post/Tabs/PostDescription.tsx [hunk 30]
```diff
@@ -25,20 +29,76 @@ const CommentsContainer = dynamic(() => import('../Comment/CommentsContainer'),
...
 
 const PostDescription: FC<IPostDescriptionProps> = (props) => {
-	const { className, canEdit, id, isEditing, toggleEdit, Sidebar, TrackerButtonComp } = props;
+	const { className, canEdit, id, isEditing, toggleEdit, Sidebar, TrackerButtonComp , allowed_roles, trackName , isOffchainPost } = props;
 	const { postData: { content, postType, postIndex, title, post_reactions } } = usePostDataContext();
-
+	const { network } = useNetworkContext();
+	const router = useRouter();
+	//write a function which redirects to the proposalType page
+	const deletePostFromUrl = (proposalType: ProposalType, trackName?: string) => {
```
Comment (alphainfinitus) [inline src/components/Post/Tabs/PostDescription.tsx:51]: This logic is already in `BackToListingView.tsx` please extract that logic into a util function and use it here and there.

`getListingPathFromProposalType(proposalType:ProposalType)`

Comment (alphainfinitus) [reply src/components/Post/Tabs/PostDescription.tsx:51]: `listingViewPath`

File: src/components/Post/Tabs/PostDescription.tsx [hunk 31]
```diff
@@ -25,20 +29,76 @@ const CommentsContainer = dynamic(() => import('../Comment/CommentsContainer'),
...
+				break;
+			case ProposalType.DEMOCRACY_PROPOSALS:
+				path = 'proposals';
+				break;
+			case ProposalType.COUNCIL_MOTIONS:
+				path = 'motions';
+				break;
+			}
+		}
+		const listingPageText = path.replace(/-|_/g, ' ');
```
Comment (alphainfinitus) [inline src/components/Post/Tabs/PostDescription.tsx:90]: listingPageText isn't needed

File: src/components/Post/Tabs/PostDescription.tsx [hunk 32]
```diff
@@ -58,6 +118,11 @@ const PostDescription: FC<IPostDescriptionProps> = (props) => {
 					{canEdit && !isEditing && <CreateOptionPoll proposalType={postType} postId={postIndex} />}
 					{TrackerButtonComp}
 					<ShareButton title={title} />
+					{
+						allowed_roles && allowed_roles.includes('moderator') && isOffchainPost && (network.includes('kusama') || network.includes('polkadot'))?
```
Comment (alphainfinitus) [inline src/components/Post/Tabs/PostDescription.tsx:122]: network is a string, use : `['polkadot', 'kusama'].includes(network)`

File: src/components/Post/ActionsBar/ReportButton/index.tsx (hunk 20 above)
Comment (alphainfinitus) [reply src/components/Post/ActionsBar/ReportButton/index.tsx:44]: instead of increasing props with `isReply, isPost` use type = 'post | comment | reply`
Please use 

`EReportType {
 POST,
 COMMENT,
 REPLY
}`

Comment (alphainfinitus) [reply src/components/Post/ActionsBar/ReportButton/index.tsx:44]: in types.ts

File: src/util/deleteContentByMod.ts
Comment (alphainfinitus) [inline src/util/deleteContentByMod.ts:1]: line 28 to 37 are not needed

**Review by alphainfinitus — CHANGES_REQUESTED** [review]
Please check comments

File: pages/api/v1/auth/actions/deleteContentByMod.ts
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/deleteContentByMod.ts:1]: let ref= postRef

if (comment) {
  ref= commentRef
}

if (comment && reply) {
  ref= replyRef
}

Comment (alphainfinitus) [reply pages/api/v1/auth/actions/deleteContentByMod.ts:1]: Please change to this as mentioned before, for readability

File: pages/api/v1/auth/actions/deleteContentByMod.ts [hunk 33]
```diff
@@ -0,0 +1,86 @@
...
+		network,
+		postId,
+		postType,
+		reason,
+		userId: String(user.id)
+	};
+	res.status(200).json({ message: 'Content deleted.' });
+
+	const triggerName = 'contentDeletedByMod';
+	fetch(`${FIREBASE_FUNCTIONS_URL}/notify`, {
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/deleteContentByMod.ts:70]: instead of .then and callback.
use await before returning function to avoid timeout

File: pages/api/v1/listing/off-chain-posts.ts
Comment (alphainfinitus) [inline pages/api/v1/listing/off-chain-posts.ts:1]: `const commentsQuerySnapshot = await postDocRef.collection('comments').count().get();`
This will need isDeleted clause too

File: pages/api/v1/posts/on-chain-post.ts
Comment (alphainfinitus) [inline pages/api/v1/posts/on-chain-post.ts:1]: Line 1010 : `const commentsSnapshot = await postDocRef.collection('comments').get();` needs isDeleted clause

Comment (alphainfinitus) [reply pages/api/v1/posts/on-chain-post.ts:1]: Line 1027

File: src/components/Post/ActionsBar/ReportButton/index.tsx (hunk 21 above)
Comment (alphainfinitus) [reply src/components/Post/ActionsBar/ReportButton/index.tsx:160]: this should be done before setLoading to true else it'll be loading forever

File: pages/api/v1/auth/actions/deleteContentByMod.ts [hunk 34]
```diff
@@ -0,0 +1,84 @@
...
+	const token = getTokenFromReq(req);
+	if(!token) return res.status(400).json({ message: 'Invalid token' });
+
+	const user = await authServiceInstance.GetUser(token);
+	if(!user) return res.status(403).json({ message: messages.UNAUTHORISED });
+
+	let ref = postsByTypeRef(network, postType)
+		.doc(String(postId));
+	if(commentId){
+		ref = ref
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/deleteContentByMod.ts:45]: this is incorrect, it'll be
ref = postsByTypeRef(network, postType)
		.doc(String(postId)).collection('comments').doc('....')....
you don't add to the same ref

also please remove console logs

File: pages/api/v1/auth/actions/deleteContentByMod.ts
Comment (sahas-01) [reply pages/api/v1/auth/actions/deleteContentByMod.ts:1]: resolved, pls check

File: pages/api/v1/auth/actions/deleteContentByMod.ts (hunk 33 above)
Comment (sahas-01) [reply pages/api/v1/auth/actions/deleteContentByMod.ts]: resolved, pls check

File: pages/api/v1/listing/off-chain-posts.ts
Comment (sahas-01) [reply pages/api/v1/listing/off-chain-posts.ts:1]: added, pls check

File: pages/api/v1/posts/on-chain-post.ts
Comment (sahas-01) [reply pages/api/v1/posts/on-chain-post.ts:1]: noted, resolved, pls check

File: pages/api/v1/auth/actions/deleteContentByMod.ts (hunk 34 above)
Comment (sahas-01) [reply pages/api/v1/auth/actions/deleteContentByMod.ts:45]: resolved, pls check

File: src/components/Post/Comment/EditableCommentContent.tsx (hunk 23 above)
Comment (sahas-01) [reply src/components/Post/Comment/EditableCommentContent.tsx:333]: done, pls check

File: src/components/Post/Comment/EditableCommentContent.tsx (hunk 25 above)
Comment (sahas-01) [reply src/components/Post/Comment/EditableCommentContent.tsx]: resolved, pls check

File: src/components/Post/Comment/EditableReplyContent.tsx (hunk 26 above)
Comment (sahas-01) [reply src/components/Post/Comment/EditableReplyContent.tsx]: done, pls check

File: src/components/Post/Comment/EditableReplyContent.tsx
Comment (sahas-01) [reply src/components/Post/Comment/EditableReplyContent.tsx:1]: done , pls check

File: src/components/Post/Tabs/PostDescription.tsx (hunk 27 above)
Comment (sahas-01) [reply src/components/Post/Tabs/PostDescription.tsx]: resolved, pls check

File: src/components/Post/Tabs/PostDescription.tsx [hunk 35]
```diff
@@ -38,7 +43,33 @@ interface IPostDescriptionProps {
 const PostDescription: FC<IPostDescriptionProps> = (props) => {
 	const { className, canEdit, id, isEditing, toggleEdit, Sidebar, TrackerButtonComp } = props;
 	const { postData: { content, postType, postIndex, title, post_reactions } } = usePostDataContext();
-
+	const { allowed_roles } = useUserDetailsContext();
+	const { network } = useNetworkContext();
+	const router = useRouter();
+	const isOffchainPost: Boolean  = postType == ProposalType.DISCUSSIONS || postType == ProposalType.GRANTS;
+	//write a function which redirects to the proposalType page
+	const listingViewPath = (proposalType: ProposalType) => {
```
Comment (alphainfinitus) [inline src/components/Post/Tabs/PostDescription.tsx:51]: listingViewPath -> goToListingViewPath

**Review by alphainfinitus — APPROVED** [review]

File: src/components/Post/Comment/Comment.tsx [hunk 36]
```diff
@@ -45,7 +46,8 @@ interface ICommentProps {
 
 export const Comment: FC<ICommentProps> = (props) => {
 	const { className, comment } = props;
-	const { user_id, content, created_at, id, replies, updated_at ,sentiment,comment_source='polkassembly', history ,spam_users_count, profile } = comment;
+	const { user_id, content, created_at, id, replies, updated_at ,sentiment,comment_source='polkassembly', history ,spam_users_count, profile, isDeleted } = comment;
+	console.log('history',history);
```
Comment (alphainfinitus) [inline src/components/Post/Comment/Comment.tsx:50]: please remove console log

File: src/types.ts [hunk 37]
```diff
@@ -281,6 +288,7 @@ export interface NetworkEvent {
 export interface ICommentHistory {
   content: string,
   created_at: Date,
+  isDeleted?: boolean,
```
Comment (alphainfinitus) [inline src/types.ts:291]: please remove isDeleted from comment history

File: src/components/Post/Comment/Comment.tsx (hunk 36 above)
Comment (sahas-01) [reply src/components/Post/Comment/Comment.tsx]: resolved, pls check

File: src/types.ts (hunk 37 above)
Comment (sahas-01) [reply src/types.ts]: resolved, pls check

**Review by alphainfinitus — APPROVED** [review]
